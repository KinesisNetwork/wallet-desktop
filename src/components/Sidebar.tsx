import * as React from 'react'
import { NavLink } from 'react-router-dom'

import { RootRoutes } from '@types'

let logo
if (process.env.IS_WEB) {
  logo = require('../icons/png/64x64.png')
}

export const Sidebar: React.SFC = () => (
  <aside className="sidebar tile is-ancestor vertical-spaced">
    <div className="sidebar-upper tile">
      <div className="tile is-vertical">
        <div className="sidebar-header tile is-parent is-vertical">
          <div className="tile is-child has-text-centered">
            <img src={logo ? logo : './icons/png/64x64.png'} className="sidebar-logo" alt="Logo alpha" />
            <p className="has-text-primary">
              100.00 KINESIS
            </p>
          </div>
          <div className="sidebar-dropdown tile is-child">
            <div className="dropdown is-hoverable">
              <div className="dropdown-trigger">
                <button aria-controls="dropdown-menu" aria-haspopup="true" className="dropdown-button button is-radiusless is-fullwidth">
                  <span className="icon is-large"><i className="fal fa-lg fa-user-circle" /></span>
                  <p>Samuello B</p>
                  <span className="icon is-small"><i className="fal fa-lg fa-angle-down" aria-hidden="true" /></span>
                </button>
              </div>

              <div className="dropdown-menu is-paddingless" id="dropdown-menu" role="menu">
                <div className="dropdown-content is-paddingless">
                  <div className="dropdown-item is-paddingless is-active">
                    <a className="button is-radiusless is-fullwidth is-shadowless">
                      <span className="icon">
                        <i className="fal fa-lg fa-check-circle" />
                      </span>
                      <span>Account 1</span>
                    </a>
                  </div>
                  <div className="dropdown-item is-paddingless">
                    <a className="button is-radiusless is-fullwidth is-shadowless">
                      <span className="icon" />
                      <span>Account 2</span>
                    </a>
                  </div>
                  <div className="dropdown-item is-paddingless">
                    <a className="button is-radiusless is-fullwidth is-shadowless">
                      <span className="icon" />
                      <span>Account 3</span>
                    </a>
                  </div>
                  
                  <hr className="dropdown-divider is-marginless" />
                  
                  <div className="dropdown-item is-paddingless">
                    <a className="button is-radiusless is-fullwidth is-shadowless">
                      <span className="icon">
                        <i className="fal fa-lg fa-plus-circle" />
                      </span>
                      <span>Add Account</span>
                    </a>
                  </div>
                  
                  <hr className="dropdown-divider is-marginless" />
                  
                  <div className="dropdown-item is-paddingless">
                    <a className="button is-radiusless is-fullwidth is-shadowless">
                      <span className="icon">
                        <i className="fal fa-lg fa-download" />
                      </span>
                      <span>Import Account</span>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        
        <div className="sidebar-list tile is-parent is-vertical">
          <SidebarItem title="Portfolio" to={RootRoutes.dashboard} icon="chart-pie" />
          <SidebarItem title="Funds" to="/funds" icon="wallet" />
          <SidebarItem title="Address Book" to={RootRoutes.addressBook} icon="address-book" />
          <SidebarItem title="Exchange" to="/exchange" icon="exchange" />
        </div>
      
      </div>
    </div>
    <div className="sidebar-lower tile is-parent">
      <div className="sidebar-lower-icon-list tile is-child">
        <a className="sidebar-lower-icon button is-shadowless" href="#" title="settings">
          <span className="icon is-large">
            <i className="fal fa-lg fa-sliders-v" />
          </span>
        </a>
        <a className="sidebar-lower-icon button is-shadowless" href="#" title="help">
          <span className="icon is-large">
            <i className="fal fa-lg fa-question-circle" />
          </span>
        </a>
        <a className="sidebar-lower-icon button is-shadowless" href="/" title="logout">
          <span className="icon is-large">
            <i className="fal fa-lg fa-sign-out-alt" />
          </span>
        </a>
      </div>
    </div>
  </aside>
)

const SidebarItem: React.SFC<{ title: string, to: string, icon: string }> = ({ icon, title, to }) => (
  <div className="sidebar-item tile is-child">
    <NavLink activeClassName="is-active" className="button is-radiusless is-fullwidth is-shadowless" to={to}>
      <div className="icon is-medium">
        <i className={`fal fa-lg fa-${icon}`} />
      </div>
      <div>{title}</div>
    </NavLink>
  </div>

)