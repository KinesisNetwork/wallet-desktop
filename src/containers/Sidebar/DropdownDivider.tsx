import * as React from 'react'

export const DropdownDivider: React.SFC = () => (
  <hr className="dropdown-divider is-marginless" />
)
