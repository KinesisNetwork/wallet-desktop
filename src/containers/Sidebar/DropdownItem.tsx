import * as React from 'react'

interface Props {
  icon?: string,
  isActive?: boolean,
  onClick?: () => void,
  subtitle?: string,
  title: string,
  titleClassName?: string
}

export const DropdownItem: React.SFC<Props> = ({ icon, isActive = false, onClick, subtitle, title, titleClassName }) => (
  <div
    className={`dropdown-item is-paddingless ${isActive && 'is-active'}`}
    onClick={onClick}
  >
    <a className="button is-radiusless is-fullwidth is-shadowless">
      <span className="icon">
        {isActive ?
          <i className="fal fa-lg fa-check-circle" /> : icon ?
          <i className={`fal fa-lg fa-${icon}`} /> : null
        }
      </span>
      <div className="dropdown-item-text vertical-spaced">
        <p className={titleClassName}>
          {title}
        </p>
        {subtitle && (
          <p className="is-size-7">
            {subtitle}
          </p>
        )}
      </div>
    </a>
  </div>
)