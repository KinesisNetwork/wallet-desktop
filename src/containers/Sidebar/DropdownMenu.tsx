import * as React from 'react'

interface Props {
  children: JSX.Element[]
}
export const DropdownMenu: React.SFC<Props> = ({ children }) => (
  <div className="dropdown-menu is-paddingless" id="dropdown-menu" role="menu">
    <div className="dropdown-content is-paddingless">
      {children}
    </div>
  </div>
)
