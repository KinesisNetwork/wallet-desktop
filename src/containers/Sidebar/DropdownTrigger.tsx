import * as React from 'react'

interface Props {
  isExpanded: boolean
  onClick: () => any
  title: string
}

export const DropdownTrigger: React.SFC<Props> = ({ isExpanded, onClick, title }) => (
  <div className={`dropdown-trigger ${isExpanded && 'is-active' }`}>
    <button
      aria-controls="dropdown-menu"
      aria-haspopup="true"
      className="dropdown-button button is-radiusless is-fullwidth"
      onClick={onClick}
    >
      <span className="icon is-large"><i className="fal fa-lg fa-user-circle" /></span>
      <p>{title}</p>
      <span className="icon">
        <i className="fal fa-lg fa-angle-down" aria-hidden="true" />
      </span>
    </button>
  </div>
)
