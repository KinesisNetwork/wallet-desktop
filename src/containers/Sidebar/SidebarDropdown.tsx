import * as React from 'react'

interface InjectedDropdownProps {
  isExpanded: boolean
  handleToggle: () => void
}
interface Props {
  children(props: InjectedDropdownProps): JSX.Element
}
interface State {
  isExpanded: boolean
}
export class SidebarDropdown extends React.PureComponent<Props, State> {
  state = { isExpanded: false }

  handleToggle = () => {
    this.setState(prevState => ({ isExpanded: !prevState.isExpanded }))
  }

  render() {
    const { isExpanded } = this.state
    return (
      <div className="sidebar-dropdown tile is-child">
        <div className={`dropdown ${isExpanded ? 'is-active' : ''}`}>
          {this.props.children({ isExpanded, handleToggle: this.handleToggle })}
        </div>
      </div>
    )
  }
}

