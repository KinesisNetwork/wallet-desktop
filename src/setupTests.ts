import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
