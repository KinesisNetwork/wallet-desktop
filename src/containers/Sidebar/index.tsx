import * as React from 'react'

import { addNextAccountFromSeedphrase, setActiveAccount } from '@actions'
import { DropdownDivider } from '@containers/Sidebar/DropdownDivider'
import { DropdownItem } from '@containers/Sidebar/DropdownItem'
import { DropdownMenu } from '@containers/Sidebar/DropdownMenu'
import { DropdownTrigger } from '@containers/Sidebar/DropdownTrigger'
import { SidebarDropdown } from '@containers/Sidebar/SidebarDropdown'
import { SidebarItem } from '@containers/Sidebar/SidebarItem'
import { SidebarLower, SidebarLowerItem } from '@containers/Sidebar/SidebarLower'
import { SidebarUpper } from '@containers/Sidebar/SidebarUpper'
import { getActiveAccount } from '@selectors'
import { RootState } from '@store'
import { PersistedAccount, RootRoutes } from '@types'
import { push } from 'connected-react-router'
import { connect } from 'react-redux'
import { RouteComponentProps, withRouter } from 'react-router'

let logo
if (process.env.IS_WEB) {
  logo = require('../../icons/png/64x64.png')
}

export const mapStateToProps = ({ wallet }: RootState) => ({
  activeAccount: getActiveAccount(wallet),
  accounts: wallet.persisted.createdAccounts,
})

export const mapDispathToProps = { addNextAccountFromSeedphrase, setActiveAccount, push }

type Props = typeof mapDispathToProps &
  ReturnType<typeof mapStateToProps> &
  RouteComponentProps<any>

export class SidebarPresentation extends React.Component<Props> {
  public dropdownAccounts = handleToggle =>
    this.props.accounts.map(a => (
      <DropdownItem
        key={a.name}
        titleClassName="has-text-weight-bold"
        title={a.name}
        isActive={this.props.activeAccount.name === a.name}
        isImport={a.imported}
        onClick={() => {
          this.setAccount(a)
          handleToggle()
        }}
      />
    ))

  public setAccount = (account: PersistedAccount) => {
    this.props.setActiveAccount({ targetAccount: account, accounts: this.props.accounts })
  }

  public addAccount = () => this.props.addNextAccountFromSeedphrase()
  public importAccount = () => this.props.push(RootRoutes.importAccount)

  render() {
    return (
      <nav className="sidebar tile is-ancestor is-vertical">
        <SidebarUpper>
          <div className="sidebar-header tile is-parent is-vertical">
            <div className="tile is-child has-text-centered">
              <img
                src={logo ? logo : './icons/png/64x64.png'}
                className="sidebar-logo"
                alt="Logo alpha"
              />
              <p className="has-text-primary">100.00 KINESIS</p>
            </div>

            <SidebarDropdown>
              {({ isExpanded, handleToggle }) => (
                <React.Fragment>
                  <DropdownTrigger
                    key="trigger"
                    isExpanded={isExpanded}
                    onClick={handleToggle}
                    title={this.props.activeAccount.name}
                  />
                  <DropdownMenu key="menu">
                    <React.Fragment>{this.dropdownAccounts(handleToggle)}</React.Fragment>
                    <DropdownDivider />
                    <DropdownItem
                      title="Add Account"
                      icon="plus-circle"
                      onClick={this.addAccount}
                    />
                    <DropdownDivider />
                    <DropdownItem
                      title="Import Account"
                      icon="download"
                      onClick={this.importAccount}
                    />
                  </DropdownMenu>
                </React.Fragment>
              )}
            </SidebarDropdown>
          </div>

          <div className="sidebar-list tile is-parent is-vertical">
            <SidebarItem title="My Money" to={RootRoutes.dashboard} icon="wallet" />
            <SidebarItem title="Contacts" to={RootRoutes.contacts} icon="address-book" />
            <SidebarItem title="Exchange" to="/exchange" icon="chart-line" />
          </div>
        </SidebarUpper>
        <SidebarLower>
          <SidebarLowerItem icon="sliders-v" hint="settings" to={RootRoutes.settings} />
          <SidebarLowerItem icon="question-circle" hint="help" to="#" />
          <SidebarLowerItem icon="sign-out-alt" hint="logout" to={RootRoutes.dashboard} />
        </SidebarLower>
      </nav>
    )
  }
}

export const Sidebar = withRouter(
  connect(
    mapStateToProps,
    mapDispathToProps,
  )(SidebarPresentation),
)
