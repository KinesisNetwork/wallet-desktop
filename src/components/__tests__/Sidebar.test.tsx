import { shallow } from 'enzyme'
import * as React from 'react'

import { Sidebar } from '@components/Sidebar'
import '../../setupTests'

describe('Sidebar', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Sidebar />)

    expect(wrapper).toMatchSnapshot()
  })
})
