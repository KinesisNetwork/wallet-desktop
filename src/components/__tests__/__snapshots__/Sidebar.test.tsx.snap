// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render correctly 1`] = `
<aside
  className="sidebar tile is-ancestor vertical-spaced"
>
  <div
    className="sidebar-upper tile"
  >
    <div
      className="tile is-vertical"
    >
      <div
        className="sidebar-header tile is-parent is-vertical"
      >
        <div
          className="tile is-child has-text-centered"
        >
          <img
            alt="Logo alpha"
            className="sidebar-logo"
            src="./icons/png/64x64.png"
          />
          <p
            className="has-text-primary"
          >
            100.00 KINESIS
          </p>
        </div>
        <div
          className="sidebar-dropdown tile is-child"
        >
          <div
            className="dropdown is-hoverable"
          >
            <div
              className="dropdown-trigger"
            >
              <button
                aria-controls="dropdown-menu"
                aria-haspopup="true"
                className="dropdown-button button is-radiusless is-fullwidth"
              >
                <span
                  className="icon is-large"
                >
                  <i
                    className="fal fa-lg fa-user-circle"
                  />
                </span>
                <p>
                  Samuello B
                </p>
                <span
                  className="icon is-small"
                >
                  <i
                    aria-hidden="true"
                    className="fal fa-lg fa-angle-down"
                  />
                </span>
              </button>
            </div>
            <div
              className="dropdown-menu is-paddingless"
              id="dropdown-menu"
              role="menu"
            >
              <div
                className="dropdown-content is-paddingless"
              >
                <div
                  className="dropdown-item is-paddingless is-active"
                >
                  <a
                    className="button is-radiusless is-fullwidth is-shadowless"
                  >
                    <span
                      className="icon"
                    >
                      <i
                        className="fal fa-lg fa-check-circle"
                      />
                    </span>
                    <span>
                      Account 1
                    </span>
                  </a>
                </div>
                <div
                  className="dropdown-item is-paddingless"
                >
                  <a
                    className="button is-radiusless is-fullwidth is-shadowless"
                  >
                    <span
                      className="icon"
                    />
                    <span>
                      Account 2
                    </span>
                  </a>
                </div>
                <div
                  className="dropdown-item is-paddingless"
                >
                  <a
                    className="button is-radiusless is-fullwidth is-shadowless"
                  >
                    <span
                      className="icon"
                    />
                    <span>
                      Account 3
                    </span>
                  </a>
                </div>
                <hr
                  className="dropdown-divider is-marginless"
                />
                <div
                  className="dropdown-item is-paddingless"
                >
                  <a
                    className="button is-radiusless is-fullwidth is-shadowless"
                  >
                    <span
                      className="icon"
                    >
                      <i
                        className="fal fa-lg fa-plus-circle"
                      />
                    </span>
                    <span>
                      Add Account
                    </span>
                  </a>
                </div>
                <hr
                  className="dropdown-divider is-marginless"
                />
                <div
                  className="dropdown-item is-paddingless"
                >
                  <a
                    className="button is-radiusless is-fullwidth is-shadowless"
                  >
                    <span
                      className="icon"
                    >
                      <i
                        className="fal fa-lg fa-download"
                      />
                    </span>
                    <span>
                      Import Account
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sidebar-list tile is-parent is-vertical"
      >
        <SidebarItem
          icon="chart-pie"
          title="Portfolio"
          to="/dashboard"
        />
        <SidebarItem
          icon="wallet"
          title="Funds"
          to="/funds"
        />
        <SidebarItem
          icon="address-book"
          title="Address Book"
          to="/address-book"
        />
        <SidebarItem
          icon="exchange"
          title="Exchange"
          to="/exchange"
        />
      </div>
    </div>
  </div>
  <div
    className="sidebar-lower tile is-parent"
  >
    <div
      className="sidebar-lower-icon-list tile is-child"
    >
      <a
        className="sidebar-lower-icon button is-shadowless"
        href="#"
        title="settings"
      >
        <span
          className="icon is-large"
        >
          <i
            className="fal fa-lg fa-sliders-v"
          />
        </span>
      </a>
      <a
        className="sidebar-lower-icon button is-shadowless"
        href="#"
        title="help"
      >
        <span
          className="icon is-large"
        >
          <i
            className="fal fa-lg fa-question-circle"
          />
        </span>
      </a>
      <a
        className="sidebar-lower-icon button is-shadowless"
        href="/"
        title="logout"
      >
        <span
          className="icon is-large"
        >
          <i
            className="fal fa-lg fa-sign-out-alt"
          />
        </span>
      </a>
    </div>
  </div>
</aside>
`;
